.github/SOLVERS.md

```diff
@@ -0,0 +1,110 @@
+MOSEK
+-----
+1) Download MOSEK as an archive from https://mosek.com/resources/downloads 
+and save to /opt/. 
+2) Apply for a licence at: https://mosek.com/resources/trial-license
+3) You will receive an email containing your "mosek.lic" file. On Linux,
+place this in `/home/<userid>/mosek/mosek.lic`
+4) Navigate to the directory where the tar.bz2 was downloaded (/opt/) and 
+enter in a shell to extract the archive: 
+````sh
+$ tar xvjf <filename>.tar.bz2
+````
+5) In the MATLAB command window, enter: 
+pathtool
+6) Click "Add with Subfolders..." and select the directory where Mosek was 
+extracted.
+
+tomlab_snopt and tomlab_cplex
+------------
+1) Download TOMLAB /SNOPT and TOMLAB /CPLEX from 
+http://tomopt.com/scripts/register.php
+2) In a terminal window, navigate to the download directory of tomlab and 
+do the following:
+````sh
+$ chmod +x <filename>.bin
+$ sudo ./<filename>.bin
+````
+3) Follow the installation instructions
+4) Place tomlab.lic and tomlab.dat in the root directory of tomlab 
+(e.g.: /opt/tomlab)
+5) In the MATLAB command window, enter: 
+pathtool
+6) Click "Add with Subfolders..." and select the directory where the 
+solvers were installed (/opt/tomlab).
+
+Gurobi
+------
+1) Register and log in to http://www.gurobi.com/
+2) Request licence from http://www.gurobi.com/downloads/download-center
+3) Download from http://www.gurobi.com/downloads/gurobi-optimizer
+4) In a shell, navigate to the directory where Gurobi was downloaded and 
+enter:
+````sh
+$ tar xvzf <archive>.tar.gz
+$ mv gurobi<ver> /opt/gurobi<ver>/
+$ cd /opt/gurobi<ver>/linux64/bin/
+$ ./grbgetkey YOUR-LICENCE-KEY-FROM-SITE
+````
+
+5) Use default directory for licence file (/home/<userid>).
+6) Edit the bash settings to include paths:
+````sh
+$ sudo nano ~/.bashrc
+````
+
+7) Append these lines, save and exit:
+export GUROBI_HOME=/usr/local/bin/gurobi<ver>/linux64
+export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/bin/gurobi<ver>/linux64/lib
+export GRB_LICENSE_FILE=/home/<userid>/gurobi.lic
+
+8) Verify it's successfully installed:
+````sh
+$ ./grbprobe
+````
+
+should give something similar to:
+````
+HOSTNAME=...
+HOSTID=...
+PLATFORM=linux64
+USERNAME=...
+SOCKETS=...
+CPU=...
+````
+
+IBM CPLEX
+----------
+1) Acquire CPLEX installation binary from [here](https://www-01.ibm.com/software/websphere/products/optimization/cplex-studio-community-edition/)
+2) Add execute permission to binary:
+````sh 
+$ chmod +x <cplexbinary>.bin
+````
+3) Execute the installer binary as superuser:
+````sh
+$ sudo ./<cplexbinary>.bin
+````
+4) Follow the installation procedure. Accept the default installation path.
+5) In the Matlab command window, type "pathtool" and hit return.
+6) Click "Add with Subfolders..." and select 
+/opt/ibm/ILOG/CPLEX_Studio<ver>/cplex/matlab.
+
+
+
+Troubleshooting:
+----------------
+
+Install GCC/G++ 4.9
```

Comment:
For OPTI and lpsolve, because I couldn't get the binaries for Linux (can't compile them on Linux for the time being, though) - Matlab only supports GCC/G++ 4.9 for MEX compilation.